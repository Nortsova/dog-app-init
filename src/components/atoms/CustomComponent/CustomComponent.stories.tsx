import type { Meta, StoryObj } from '@storybook/react';
import { CustomComponent } from '.';

const meta = {
  title: 'Atoms/CustomComponent',
  component: CustomComponent,
  argTypes: {
    status: {
      options: ['error', 'success', 'loading'],
      control: 'radio'
    }
  }
} satisfies Meta<typeof CustomComponent>;

export default meta;
type Story = StoryObj<typeof meta>;

export const ErrorCustomComponent: Story = {
  args: {
    status: 'error'
  }
};
export const SuccessCustomComponent: Story = {
  args: {
    status: 'success'
  }
};

export const LoadingCustomComponent: Story = {
  args: {
    status: 'loading'
  }
};

export const DefaultCustomComponent: Story = {
  args: {}
};
