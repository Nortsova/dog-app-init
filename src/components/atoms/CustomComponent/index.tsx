import { FC } from 'react';

type CustomComponentProps = {
  status?: 'error' | 'success' | 'loading';
};

export const CustomComponent: FC<CustomComponentProps> = ({ status }) => {
  switch (status) {
    case 'error':
      return <div>Error</div>;
    case 'success':
      return <div>Success</div>;
    case 'loading':
      return <div>Loading...</div>;
    default:
      return <div>Default</div>;
  }
};
