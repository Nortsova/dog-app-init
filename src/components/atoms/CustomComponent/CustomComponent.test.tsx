import { composeStories } from '@storybook/react';
import * as stories from './CustomComponent.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('Custom Component', () => {
  const {
    ErrorCustomComponent,
    SuccessCustomComponent,
    LoadingCustomComponent,
    DefaultCustomComponent
  } = composeStories(stories);
  testSnapshot(<ErrorCustomComponent />);
  testSnapshot(<SuccessCustomComponent />);
  testSnapshot(<LoadingCustomComponent />);
  testSnapshot(<DefaultCustomComponent />);
});
