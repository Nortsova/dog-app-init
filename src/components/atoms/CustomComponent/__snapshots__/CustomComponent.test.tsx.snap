// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Component should match snapshot Component 1`] = `
<div>
  <div>
    Error
  </div>
</div>
`;

exports[`Custom Component should match snapshot Component 2`] = `
<div>
  <div>
    Success
  </div>
</div>
`;

exports[`Custom Component should match snapshot Component 3`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;

exports[`Custom Component should match snapshot Component 4`] = `
<div>
  <div>
    Default
  </div>
</div>
`;
